response_operations_ui/views/admin/manage_user_accounts.py

```diff
@@ -66,7 +72,171 @@ def manage_user_accounts():
+            except HTTPError:
+                flash(f"Failed add [{permission}] to the user, please try again", "error")
+                return redirect(url_for("admin_bp.manage_account", user=user["emails"][0]["value"]))
+            was_permission_changed = True
+
+        if translated_permission in user_groups:
+            try:
+                remove_group_membership(user_id, group_id)  # Not ticked but in group, need to remove it
+            except HTTPError:
+                flash(f"Failed remove [{permission}] to the user, please try again", "error")
+                return redirect(url_for("admin_bp.manage_account", user=user["emails"][0]["value"]))
+            was_permission_changed = True
+        continue  # Nothing to do, already not in the group
+
+    if was_permission_changed:
+        try:
+            NotifyController().request_to_notify(
+                email=user["emails"][0]["value"],
+                template_name="update_user_permissions",
+                personalisation={},
+            )
+        except NotifyError as e:
+            logger.error("failed to send email", msg=e.description, exc_info=True)
+            flash("Failed to send email, please try again", "error")
+            return redirect(url_for("admin_bp.manage_account", user=user["emails"][0]["value"]))
+
+        flash("User account has been successfully changed. An email to inform the user has been sent.")
+
+    return redirect(url_for("admin_bp.manage_user_accounts"))
+
+
+@admin_bp.route("/manage-account/<user_id>/delete", methods=["GET"])
+@login_required
+def get_delete_uaa_user(user_id):
+    _verify_user_admin_permission()
+
+    if user_id == session["user_id"]:
+        flash("You cannot delete your own user account", "error")
+        logger.error("User tried to delete themselves", user_id=user_id)
+        return redirect(url_for("admin_bp.manage_user_accounts"))
+
+    user = get_user_by_id(user_id)
+    if user is None:
+        flash("User does not exist", "error")
+        return redirect(url_for("admin_bp.manage_user_accounts"))
+
+    name = user["name"]["givenName"] + " " + user["name"]["familyName"]
+    email = user["emails"][0]["value"]
+    return render_template("admin/user-delete.html", name=name, email=email)
+
+
+@admin_bp.route("/manage-account/<user_id>/delete", methods=["POST"])
+@login_required
+def post_delete_uaa_user(user_id):
+    _verify_user_admin_permission()
+
+    if user_id == session["user_id"]:
+        flash("You cannot delete your own user account", "error")
+        logger.error("User tried to delete themselves", user_id=user_id)
+        return redirect(url_for("admin_bp.manage_user_accounts"))
+
+    user = get_user_by_id(user_id)
+    if user is None:
+        flash("User does not exist", "error")
+        return redirect(url_for("admin_bp.manage_user_accounts"))
+
+    try:
+        NotifyController().request_to_notify(
```

Comment:
I did it this way to deal with the worst case.  It'd be better to be informed of a deletion and have it not happen instead of having it deleted and never get told about it.

Reply:
Fair point, my thinking was if whatever reason Notify became a blocker it doesn't stop us still being able to perform the action. It doesn't matter a great deal so can be left as is